pkg/resources/view.go

```diff
@@ -237,63 +622,130 @@ func UpdateView(d *schema.ResourceData, meta interface{}) error {
 		if comment := d.Get("comment").(string); comment == "" {
 			err := client.Views.Alter(ctx, sdk.NewAlterViewRequest(id).WithUnsetComment(true))
 			if err != nil {
-				return fmt.Errorf("error unsetting comment for view %v", d.Id())
+				return diag.FromErr(fmt.Errorf("error unsetting comment for view %v", d.Id()))
 			}
 		} else {
 			err := client.Views.Alter(ctx, sdk.NewAlterViewRequest(id).WithSetComment(comment))
 			if err != nil {
-				return fmt.Errorf("error updating comment for view %v", d.Id())
+				return diag.FromErr(fmt.Errorf("error setting comment for view %v", d.Id()))
 			}
 		}
 	}
 
 	if d.HasChange("is_secure") {
-		if d.Get("is_secure").(bool) {
-			err := client.Views.Alter(ctx, sdk.NewAlterViewRequest(id).WithSetSecure(true))
+		if v := d.Get("is_secure").(string); v != BooleanDefault {
+			parsed, err := booleanStringToBool(v)
 			if err != nil {
-				return fmt.Errorf("error setting secure for view %v", d.Id())
+				return diag.FromErr(err)
+			}
+			if parsed {
+				err = client.Views.Alter(ctx, sdk.NewAlterViewRequest(id).WithSetSecure(parsed))
+			} else {
+				err = client.Views.Alter(ctx, sdk.NewAlterViewRequest(id).WithUnsetSecure(parsed))
```

Comment:
I think it's incorrect, this whole if statement should be removed and only 
```go
err = client.Views.Alter(ctx, sdk.NewAlterViewRequest(id).WithSetSecure(parsed))
```
should be called. Unset should be reserved for setting default value.

Reply:
🤦 Fixed.